mantidproject/mantid
# Merge pull request #27841 from mantidproject/27077_ability_to_cancel_live_data

This was due to:
1. Too long a default UpdateEvery default
2. Too infrequest progress updates from MonitorLiveData
3. The Algorithm Details box not adding an algorithm until it sends a progress update
4. The StartLiveData custom Dialog not showing the default for UpdateEvery

Fixes:
1. Decreased the default UpdateEvery from 60 to 30 seconds
2. Change the default timeout for FakeISISHistongramDAE from 30 to 60s
3. improved the progress logging for MonitorLivedata to update more frequently with an idle counter as well.
4. Changed the tie function used in custom algorithm dialogs to set the placeholder text for line edits
5. Changed the log level for user cancelling of algorithms from Error to warning so it does not trigger the notification service
6. Changed the log message for user cancellation from Terminated to cancelled to reduce confusion




**Report to:** @mducle 


**To test:**

1. Start workbench
1. File->settings, change your facility to TEST_LIVE, instrument to 
1. Run FakeISISHistogramDAE with the defaults
1. Load->Live Data
   1. Instrument ISIS_Histgram
   1. Connection histo
   1. Start: Now, No Processing or post processing
   1. Set a name for the output workspace
   1. If you do not have a history value for UpdateEvery it should have a placeholder text of 30, if you do, delete your value and the placeholder should be visible
   1. run start live data
1. Click the detail button on the algorithms widget
   1. You should get two entries one for Fake.... and one for MontiorLiveData
   1. Check that the update text makes sense
1. Cancel MonitorLiveData
   1. It should cancel within a reasonable amount of time
   1. There should not be a notiifcation pop up saying there are errors in the log
   1. The message in the log should not mention termination

1. Also check other custom algorithm Dialogs and make sure that the placeholder text does not appear in a bad way, not all of these might be affected.
   1. SliceMD
   1. BinMD
   1. CatalogPublish (you will need CatalogLogin first)
   1. ConvertTableToMatrixWorkspace
   1. Fit
   1. Load
   1. SmoothNeighbours
   1. SortTableWorkspace
   1. PlotAsymmetryByLogValue

Fixes #27077

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
